# HG changeset patch
# User Rik <rik@octave.org>
# Date 1534367608 25200
#      Wed Aug 15 14:13:28 2018 -0700
# Node ID b785394f10d03ba1bdac6096586cd4eecc2933b5
# Parent  95eb72d50fb0cd5386dd5098225e5c790ceefa53
shg.m: Don't create a figure if none exists.

* shg.m: Redo docstring.  Use the "currentfigure" of the root object, rather
than gcf, to get the current figure so as not to create one if one does not
exist.  Change the "visible" attribute of the figure to "on" unconditionally.

* figure.m: Add shg to list of @seealso links in docstring.  Use figure handle
'f' rather than gcf when calling __show_figure__ since it is already known.

diff --git a/scripts/plot/util/figure.m b/scripts/plot/util/figure.m
--- a/scripts/plot/util/figure.m
+++ b/scripts/plot/util/figure.m
@@ -34,17 +34,17 @@
 ## Multiple property-value pairs may be specified for the figure object, but
 ## they must appear in pairs.
 ##
 ## The optional return value @var{h} is a graphics handle to the created figure
 ## object.
 ##
 ## Programming Note: The full list of properties is documented at
 ## @ref{Figure Properties,,Figure Properties}.
-## @seealso{axes, gcf, clf, close}
+## @seealso{axes, gcf, shg, clf, close}
 ## @end deftypefn
 
 ## Author: jwe, Bill Denney
 
 function h = figure (varargin)
 
   nargs = nargin;
 
@@ -104,17 +104,17 @@ function h = figure (varargin)
     set (0, "currentfigure", f);
   endif
 
   ## When switching to figure N, make figure visible and on top of stack,
   ## unless visibility is explicitly switched off.
   if (! init_new_figure && ! any (strcmpi (varargin(1:2:end), "visible")
                                   && strcmpi (varargin(2:2:end), "off")))
     set (f, "visible", "on");
-    __show_figure__ (gcf ());
+    __show_figure__ (f);
   endif
 
   if (nargout > 0)
     h = f;
   endif
 
 endfunction
 
diff --git a/scripts/plot/util/shg.m b/scripts/plot/util/shg.m
--- a/scripts/plot/util/shg.m
+++ b/scripts/plot/util/shg.m
@@ -15,23 +15,31 @@
 ## You should have received a copy of the GNU General Public License
 ## along with Octave; see the file COPYING.  If not, see
 ## <https://www.gnu.org/licenses/>.
 
 ## -*- texinfo -*-
 ## @deftypefn {} {} shg
 ## Show the graph window.
 ##
-## Currently, this is the same as executing @code{drawnow}.
-## @seealso{drawnow, figure}
+## This function makes the current figure visible, and places it on top of
+## of all other plot windows.
+##
+## Programming Note: @code{shg} is equivalent to @code{figure (gcf)} assuming
+## that a current figure exists.
+## @seealso{figure, drawnow, gcf}
 ## @end deftypefn
 
 ## Author: jwe
 
 function shg ()
 
   if (nargin != 0)
     warning ("shg: ignoring extra arguments");
   endif
 
-  __show_figure__ (gcf ());
+  hf = get (0, "currentfigure");
+  if (! isempty (hf))
+    set (hf, "visible", "on");
+    __show_figure__ (hf));
+  endif
 
 endfunction
